atc/db/check_factory.go

```diff
@@ -139,8 +139,13 @@ func (c *checkFactory) Resources() ([]Resource, error) {
 	var resources []Resource
 
 	rows, err := resourcesQuery.
-		Join("(select DISTINCT(resource_id) FROM job_inputs) ji ON ji.resource_id = r.id").
+		LeftJoin("(select DISTINCT(resource_id) FROM job_inputs) ji ON ji.resource_id = r.id").
+		LeftJoin("(select DISTINCT(resource_id) FROM job_outputs) jo ON jo.resource_id = r.id").
 		Where(sq.Eq{"p.paused": false}).
+		// filter for resources that are job inputs
+		Where(sq.NotEq{"ji.resource_id": nil}).
+		// also inlude put-only resources that have errored
+		Suffix("OR (b.status IN ('aborted','failed','errored') AND ji.resource_id IS NULL)").
```

Comment:
I wanted to avoid doing this but I couldn't figure out a way to either 1) clear out the where and rebuild it or 2) add a where that `OR`'s instead of `AND`'s. Neither seemed possible with the `mastermind/squirrel` library

Reply:
I don't think you can clear out the `WHERE` in the `resourceQuery`, but you should be able to construct something like
```go
Where(sq.Or{
	sq.And{
		sq.Eq{"p.paused": false},
		sq.NotEq{"ji.resource_id": nil},
	},
	sq.And{
		sq.Expr("b.status IN ('aborted','failed','errored')"),
		sq.Eq{"ji.resource_id": nil},
	},
}).
```

Reply:
Yes, the issue is it will be `AND`'d with the `WHERE` in the base `resourcesQuery` var. See:

https://github.com/concourse/concourse/blob/df62ba0d8a2d4b1b2d99bff9a25bb00874e8eb6e/atc/db/resource.go#L76-L106

Thinking maybe I'll split it into two vars.

Reply:
Isn't it fine to `AND` that `Where(sq.Eq{"r.active": true})` since we only want to filter on active resources as well? My understanding is that the combined clause would look something like 
`... WHERE r.active = true AND ((p.paused = false AND ji.resource_id IS NOT NULL) OR (b.status IN ('aborted','failed','errored') AND ji.resource_id IS NOT NULL))`

We probably don't want to lidar check inactive resources anyways since an inactive resource indicates that it was "orphaned" as result of an existing pipeline config getting updated
https://github.com/concourse/concourse/blob/df62ba0d8a2d4b1b2d99bff9a25bb00874e8eb6e/atc/db/team.go#L1465-L1471
https://github.com/concourse/concourse/blob/df62ba0d8a2d4b1b2d99bff9a25bb00874e8eb6e/atc/db/team.go#L505